src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -152,6 +152,11 @@ public VisitArgumentResult(VisitResult visitResult, Optional<LocalState> stateFo
         /// </summary>
         private readonly ImmutableDictionary<BoundExpression, (NullabilityInfo Info, TypeSymbol Type)>.Builder _analyzedNullabilityMapOpt;
 
+        /// <summary>
+        /// Contains the map of calls to reinferred method symbols, used by the optional rewriter phase of the compiler.
+        /// </summary>
+        private readonly ImmutableDictionary<BoundCall, MethodSymbol>.Builder _updatedMethodSymbolMapOpt;
```

Comment:
I think this is probably what you were getting at with the original design, but in the future it might be nice to consolidate the 'maps' into a single lookup that pulls out the right, updated thing if it's there.